### PR TITLE
Use monitoring common

### PR DESCRIPTION
**Pull request by MykolaSolyanko**

### CODE REVIEW

File: monitoringcontroller/cover.html [hunk 1]
```diff
@@ -0,0 +1,183 @@
+
```
Comment (Eabramov84) [inline monitoringcontroller/cover.html:1]: remove this file

File: monitoringcontroller/monitoringcontroller.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	"context"
+
+	"github.com/aoscloud/aos_communicationmanager/cloudprotocol"
+	"github.com/aoscloud/aos_communicationmanager/config"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// AMQPSender sends monitoring data.
+type AMQPSender interface {
```
Comment (Eabramov84) [inline monitoringcontroller/monitoringcontroller.go:30]: Rename to Monitoring sender

File: monitoringcontroller/monitoringcontroller.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+// Close closes monitor controller instance.
+func (monitor *MonitorController) Close() {
+	if monitor.cancelFunction != nil {
+		monitor.cancelFunction()
+	}
+}
+
+// SendMonitoringData sends monitoring data.
+func (monitor *MonitorController) SendMonitoringData(monitoringData cloudprotocol.MonitoringData) {
```
Comment (Eabramov84) [inline monitoringcontroller/monitoringcontroller.go:78]: please control size of message which is defined in config

File: config/config.go [hunk 4]
```diff
@@ -40,9 +42,9 @@ type Crypt struct {
 
 // UMController configuration for update controller.
 type UMController struct {
-	ServerURL string           `json:"serverUrl"`
-	UMClients []UMClientConfig `json:"umClients"`
-	UpdateTTL Duration         `json:"updateTtl"`
+	ServerURL string            `json:"serverUrl"`
+	UMClients []UMClientConfig  `json:"umClients"`
+	UpdateTTL aostypes.Duration `json:"updateTtl"`
```
Comment (Eabramov84) [inline config/config.go:47]: not relevant changes. It already done on another mr

File: config/config.go [hunk 5]
```diff
@@ -52,38 +54,20 @@ type UMClientConfig struct {
...
-	EnableSystemMonitoring bool       `json:"enableSystemMonitoring"`
-	MaxOfflineMessages     int        `json:"maxOfflineMessages"`
-	SendPeriod             Duration   `json:"sendPeriod"`
-	PollPeriod             Duration   `json:"pollPeriod"`
-	RAM                    *AlertRule `json:"ram"`
-	CPU                    *AlertRule `json:"cpu"`
-	UsedDisk               *AlertRule `json:"usedDisk"`
-	InTraffic              *AlertRule `json:"inTraffic"`
-	OutTraffic             *AlertRule `json:"outTraffic"`
+	MinTimeout   aostypes.Duration `json:"minTimeout"`
```
Comment (Eabramov84) [inline config/config.go:59]: the same

File: config/config.go [hunk 6]
```diff
@@ -52,38 +54,20 @@ type UMClientConfig struct {
...
 
 // Alerts configuration for alerts.
 type Alerts struct {
-	EnableSystemAlerts bool     `json:"enableSystemAlerts"`
-	SendPeriod         Duration `json:"sendPeriod"`
-	MaxMessageSize     int      `json:"maxMessagesize"`
-	MaxOfflineMessages int      `json:"maxOfflineMessages"`
-	Filter             []string `json:"filter"`
+	EnableSystemAlerts bool              `json:"enableSystemAlerts"`
+	SendPeriod         aostypes.Duration `json:"sendPeriod"`
```
Comment (Eabramov84) [inline config/config.go:67]: the same

File: config/config.go [hunk 7]
```diff
@@ -94,12 +78,12 @@ type Migration struct {
...
-	DownloadDir            string   `json:"downloadDir"`
-	DecryptDir             string   `json:"decryptDir"`
-	MaxConcurrentDownloads int      `json:"maxConcurrentDownloads"`
-	RetryDelay             Duration `json:"retryDelay"`
-	MaxRetryDelay          Duration `json:"maxRetryDelay"`
-	DownloadPartLimit      int      `json:"downloadPartLimit"`
+	DownloadDir            string            `json:"downloadDir"`
+	DecryptDir             string            `json:"decryptDir"`
+	MaxConcurrentDownloads int               `json:"maxConcurrentDownloads"`
+	RetryDelay             aostypes.Duration `json:"retryDelay"`
```
Comment (Eabramov84) [inline config/config.go:84]: the same

File: config/config.go [hunk 8]
```diff
@@ -111,28 +95,28 @@ type SMConfig struct {
 
 // SMController SM controller configuration.
 type SMController struct {
-	SMList    []SMConfig `json:"smList"`
-	UpdateTTL Duration   `json:"updateTtl"`
+	SMList    []SMConfig        `json:"smList"`
+	UpdateTTL aostypes.Duration `json:"updateTtl"`
```
Comment (Eabramov84) [inline config/config.go:99]: the same

File: config/config.go [hunk 9]
```diff
@@ -111,28 +95,28 @@ type SMConfig struct {
...
+	CertStorage           string                `json:"certStorage"`
+	ServiceDiscoveryURL   string                `json:"serviceDiscoveryUrl"`
+	IAMServerURL          string                `json:"iamServerUrl"`
+	IAMPublicServerURL    string                `json:"iamPublicServerUrl"`
+	FileServerURL         string                `json:"fileServerUrl"`
+	CMServerURL           string                `json:"cmServerUrl"`
+	Downloader            Downloader            `json:"downloader"`
+	WorkingDir            string                `json:"workingDir"`
+	BoardConfigFile       string                `json:"boardConfigFile"`
+	UnitStatusSendTimeout aostypes.Duration     `json:"unitStatusSendTimeout"`
```
Comment (Eabramov84) [inline config/config.go:114]: the same

**Review by al1img — COMMENTED** [review]
We should have monitoring package in communicationmanager as well, it should be disabled by default. But we should have the possibility to enable it.

File: monitoringcontroller/monitoringcontroller.go (hunk 3 above)
Comment (MykolaSolyanko) [reply monitoringcontroller/monitoringcontroller.go:78]: good point

File: monitoringcontroller/monitoringcontroller.go (hunk 2 above)
Comment (MykolaSolyanko) [reply monitoringcontroller/monitoringcontroller.go:30]: Done

File: config/config.go (hunk 4 above)
Comment (MykolaSolyanko) [reply config/config.go:47]: Done

File: monitoringcontroller/cover.html (hunk 1 above)
Comment (MykolaSolyanko) [reply monitoringcontroller/cover.html]: Done

File: config/config.go (hunk 4 above)
Comment (MykolaSolyanko) [reply config/config.go]: Done

File: config/config.go (hunk 5 above)
Comment (MykolaSolyanko) [reply config/config.go]: Done

File: config/config.go (hunk 6 above)
Comment (MykolaSolyanko) [reply config/config.go]: Done

File: config/config.go (hunk 7 above)
Comment (MykolaSolyanko) [reply config/config.go]: Done

File: config/config.go (hunk 8 above)
Comment (MykolaSolyanko) [reply config/config.go]: Done

File: config/config.go (hunk 9 above)
Comment (MykolaSolyanko) [reply config/config.go]: Done

**Comment by MykolaSolyanko** [comment]
> We should have monitoring package in communicationmanager as well, it should be disabled by default. But we should have the possibility to enable it.

It will be done in separate PR

File: monitoringcontroller/monitoringcontroller.go [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+import (
+	"context"
+
+	"github.com/aoscloud/aos_common/api/cloudprotocol"
+	"github.com/aoscloud/aos_communicationmanager/config"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// MonitoringSender sends monitoring data.
```
Comment (al1img) [inline monitoringcontroller/monitoringcontroller.go:29]: Could you add headers `Types`, `Public` into this file?

File: monitoringcontroller/monitoringcontroller.go [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+type MonitoringSender interface {
+	SendMonitoringData(monitoringData cloudprotocol.MonitoringData) (err error)
+}
+
+// MonitorController instance.
+type MonitorController struct {
```
Comment (al1img) [inline monitoringcontroller/monitoringcontroller.go:35]: `monitoringcontroller.MonitorController` looks not so good. Consider renaming to just `Controller` or `Instance`.

File: monitoringcontroller/monitoringcontroller.go (hunk 10 above)
Comment (MykolaSolyanko) [reply monitoringcontroller/monitoringcontroller.go:33]: Done

File: monitoringcontroller/monitoringcontroller.go [hunk 12]
```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: Apache-2.0
```
Comment (al1img) [inline monitoringcontroller/monitoringcontroller.go:1]: Package name `monitorcontroller`?

Comment (MykolaSolyanko) [reply monitoringcontroller/monitoringcontroller.go]: Done

**Review by al1img — APPROVED** [review]
